py/dml/objects.py

```diff
@@ -216,8 +215,12 @@ def __init__(self, ident, site, parent, arraylens = (), idxvars = ()):
 
         self._confidential = False
 
-        self.uniq = len(dml.globals.objects) + 1
-        dml.globals.objects.append(self)
+        # during code generation, set to a unique positive integer,
+        # used as object ID, e.g. in the id_infos array.  This integer
+        # number is the index into object.Device.objects *plus one*;
+        # the plus one is because object ID 0 is reserved in runtime
+        # for an invalid reference.
+        self.uniq = None
```

Comment:
since uniq is not just any unique integer, but *the* object ID of the object used in runtime, perhaps we should also rename it, e.g. what about `object_id` (and `hook_id`)?

Reply:
I see your point. I would rather have a standardized name than just tacking a prefix onto `id`. The "array of things and index of thing used for identification" pattern is one I've used a lot, just search for `uniq` and you'll find tons of applications. Even more in RAII. I don't have a good name for it beyound `uniq`. `idx`? That can be confused with other things.